app/models/mixins/retirement_mixin.rb

```diff
@@ -129,15 +129,19 @@ def retire_now(requester = nil)
     if retired
       return if retired_validated?
       _log.info("#{retirement_object_title}: [#{name}], Retires On: [#{retires_on.strftime("%x %R %Z")}], was previously retired, but currently #{retired_invalid_reason}")
-    else
-      update_attributes(:retirement_requester => requester)
-      event_name = "request_#{retirement_event_prefix}_retire"
-      _log.info("calling #{event_name}")
-      begin
-        raise_retirement_event(event_name, requester)
-      rescue => err
-        _log.log_backtrace(err)
+    elsif retirement_state != "initializing"
+      lock do
+        update_attributes(:retirement_state => "initializing", :retirement_requester => requester)
+        event_name = "request_#{retirement_event_prefix}_retire"
+        _log.info("calling #{event_name}")
+        begin
+          raise_retirement_event(event_name, requester)
+        rescue => err
+          _log.log_backtrace(err)
+        end
       end
+    else
+      _log.info("#{retirement_object_title}: [#{name}] has already started retirement")
```

Comment:
@d-m-u 
Should we log what the current state is, it might help us debugging.
Can we add some spec for the retirement_state = initializing